### DISCUSSION TITLE
ScriptReference/SerializeReference

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#90] [Notes]
# ScriptReference/SerializeReference

https://docs.unity3d.com/ScriptReference/SerializeReference.html

**Comment by vertxxyz** [comment]
`SerializeReference` doesn't have a suitable drawer for `null` values, nor does it allow you to assign subtypes via the inspector.
I have an extension—[Vertx.SerializeReferenceDropdown](https://github.com/vertxxyz/Vertx.SerializeReferenceDropdown)—with that functionality that also correctly supports Property Drawers.

**Comment by vertxxyz** [comment]
You can use the `UnityEngine.Scripting.APIUpdating.MovedFromAttribute` when refactoring namespaces or assemblies and this should fixup broken references.

Using a global find and replace for the string of the previous class name also works.
